aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputFile.java

```diff
@@ -70,7 +70,7 @@ public PositionOutputStream createOrOverwrite() {
 
   @Override
   public InputFile toInputFile() {
-    return new S3InputFile(client(), uri(), awsProperties(), metrics());
+    return new S3InputFile(client(), uri(), null, awsProperties(), metrics());
```

Comment:
Similar to GCS, could we just call `S3InputFile.fromLocation(...)` to avoid calling the constructor with null?